File: modules/syslog.php
<?php


/*
 * LMS version 1.11-git
 *
 *  (C) Copyright 2001-2012 LMS Developers
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License Version 2 as
 *  published by the Free Software Foundation.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
 *  USA.
 *
 *  $Id: syslog.php, 2012/11/02 19:21, Sylwester Kondracki Exp $
 *
 */


$layout['pagetitle'] = 'Logi systemowe';

$filter = array();

if (!isset($_GET['page'])) $SESSION->restore('sl_page',$_GET['page']);

if (!isset($_GET['sl_df']))	$SESSION->restore('sl_df',$filter['df']);	else $filter['df'] = $_GET['sl_df'];	$SESSION->save('sl_df',$filter['df']);
if (!isset($_GET['sl_dt']))	$SESSION->restore('sl_dt',$filter['dt']);	else $filter['dt'] = $_GET['sl_dt'];	$SESSION->save('sl_dt',$filter['dt']);
if (!isset($_GET['sl_mod']))	$SESSION->restore('sl_mod',$filter['mod']);	else $filter['mod'] = $_GET['sl_mod'];	$SESSION->save('sl_mod',$filter['mod']);
if (!isset($_GET['sl_ev']))	$SESSION->restore('sl_ev',$filter['ev']);	else $filter['ev'] = $_GET['sl_ev'];	$SESSION->save('sl_ev',$filter['ev']);
if (!isset($_GET['sl_us']))	$SESSION->restore('sl_us',$filter['us']);	else $filter['us'] = $_GET['sl_us'];	$SESSION->save('sl_us',$filter['us']);
if (!isset($_GET['sl_dus']))	$SESSION->restore('sl_dus',$filter['dus']);	else $filter['dus'] = $_GET['sl_dus'];	$SESSION->save('sl_dus',$filter['dus']);


$syslog = $DB->GetAll('SELECT s.*, u.login FROM syslog AS s JOIN users AS u ON (u.id = s.uid) WHERE 1=1 '
    .($filter['mod'] ? ' AND s.module='.$filter['mod'] : '')
    .($filter['ev'] ? ' AND s.event='.$filter['ev'] : '')
    .($filter['us'] ? ' AND s.uid='.$filter['us'] : '')
    .($filter['df'] ? ' AND s.cdate>='.strtotime($filter['df'].' 00:00:00') : '')
    .($filter['dt'] ? ' AND s.cdate<='.strtotime($filter['dt'].' 23:59:59') : '')
    .(!$filter['dus'] ? ' AND u.deleted=0 ' : '')
    .' ORDER BY s.cdate DESC');

$page = (!isset($_GET['page']) ? 1 : $_GET['page']);
$pagelimit = (! $LMS->CONFIG['phpui']['syslog_pagelimit'] ? 100 : $LMS->CONFIG['phpui']['syslog_pagelimit']);
$start = ($page - 1) * $pagelimit;

$SESSION->save('sl_page',$page);
$SESSION->save('backto',$_SERVER['QUERY_STRING']);

$listdata['total'] = sizeof($syslog);

$SMARTY->assign('listdata',$listdata);
$SMARTY->assign('pagelimit',$pagelimit);
$SMARTY->assign('page',$page);
$SMARTY->assign('start',$start);
$SMARTY->assign('filter',$filter);
$SMARTY->assign('users',$DB->GetAll('SELECT id, login, deleted FROM users'.(!$filter['dus'] ? ' WHERE deleted=0' : '').' ORDER BY login ASC;'));
$SMARTY->assign('syslog',$syslog);

$SMARTY->display('syslog.html');

?>

File: daemon/modules/parser/extensions/syslog.c
/*
 *  LMS version 1.11-git
 *
 *  (C) Copyright 2001-2013 LMS Developers
 *
 *  Please, see the doc/AUTHORS for more information about authors!
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License Version 2 as
 *  published by the Free Software Foundation.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
 *  USA.
 *
 *  $Id$
 */

#include <stdio.h>
#include <syslog.h>

#include "syslog.h"
#include "tscript_extensions.h"
#include "tscript_variables.h"

tscript_value * tscript_ext_syslog_syslog(tscript_value *arg)
{
	tscript_value *log;
	int n = LOG_INFO;

	if( arg->type == TSCRIPT_TYPE_ARRAY )
	{
		tscript_value *index, *level;
		
		index = tscript_value_create_number(0);
		log = tscript_value_convert_to_string(tscript_value_dereference(
			*tscript_value_array_item_ref(&arg, index)));
		tscript_value_free(index);

		index = tscript_value_create_number(1);
		level = tscript_value_convert_to_number(tscript_value_dereference(
			*tscript_value_array_item_ref(&arg, index)));
		tscript_value_free(index);
		
		n = tscript_value_as_number(level);

		tscript_value_free(level);
	}
	else
	{
		log = tscript_value_convert_to_string(arg);
	}

	syslog(n, "%s", tscript_value_as_string(log));

	tscript_value_free(log);

	return tscript_value_create_null();
}

void tscript_ext_syslog_init(tscript_context *context)
{
	tscript_add_extension(context, "syslog", tscript_ext_syslog_syslog, 1, 2);

	*tscript_variable_get_reference(context, "LOG_EMERG") = tscript_value_create_number(LOG_EMERG);
	*tscript_variable_get_reference(context, "LOG_ALERT") = tscript_value_create_number(LOG_ALERT);
	*tscript_variable_get_reference(context, "LOG_CRIT") = tscript_value_create_number(LOG_CRIT);
	*tscript_variable_get_reference(context, "LOG_ERR") = tscript_value_create_number(LOG_ERR);
	*tscript_variable_get_reference(context, "LOG_WARNING") = tscript_value_create_number(LOG_WARNING);
	*tscript_variable_get_reference(context, "LOG_NOTICE") = tscript_value_create_number(LOG_NOTICE);
	*tscript_variable_get_reference(context, "LOG_INFO") = tscript_value_create_number(LOG_INFO);
	*tscript_variable_get_reference(context, "LOG_DEBUG") = tscript_value_create_number(LOG_DEBUG);
}

void tscript_ext_syslog_close(tscript_context *context)
{
	tscript_remove_extension(context, "syslog");
}
